status/status.go

```diff
@@ -66,7 +74,7 @@ func (h *healthHandlerImpl) computeNewHealthStatus(req *http.Request) (health.He
 	if sharedSecret := h.healthCheckSharedSecret.CurrentString(); sharedSecret != "" {
 		token, err := httpserver.ParseBearerTokenHeader(req)
 		if err != nil || sharedSecret != token {
-			return health.HealthStatus{}, http.StatusUnauthorized
+			return healthStatusUnauthorized, http.StatusUnauthorized
```

Comment:
My preference was to log when we hit this case but there's no accessible ctx and the `loggingHealthStatusChangeHandler` used above derives the status code from the health status rather than from the status returned by `computeNewHealthStatus`